extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -339,8 +340,9 @@ public String getName() {
                 }
             });
             if (userInfo != null) {
-                OidcUtils.setSecurityIdentityRoles(builder, resolvedContext.oidcConfig,
-                        new JsonObject(userInfo.getJsonObject().toString()));
+                var rolesJson = new JsonObject(userInfo.getJsonObject().toString());
```

Comment:
I'm not sure it is the right place, https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java#L263 has JsonObject representing the roles, so add `OidcUtils.setSecurityIdentityPermissions(builder, resolvedContext.oidcConfig, rolesJson);` there only

Reply:
I think it is the right place because you don't expect `setSecurityIdentityRoles` to set permissions, but doing it will simplify things, so I'll do as you suggest. Are you fine with renaming it to `setSecurityIdentityRolesAndPermissions`?

Reply:
@michalvavrik I did not suggest to have permissions set as part of setting roles, I'm saying that you don't need to create another JsonObject, it is already available in the linked code - and your change implies it is only set if the userinfo is available, sorry of I was not clear

Reply:
I see @sberyozkin , but that's https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java#L263 you pointed to is different flow. Here I just mirrow what setting roles does. and I set parameter value to the variable just to not create jsonobject twice. In that case, it seems correct to me.

Reply:
Oh, sorry, typical, I got confused again :-)

Reply:
np